### PR TITLE
Button `isLoading` prop

### PR DESCRIPTION
**Pull request by lin-jessica**
Add loading state to filled button components
- `isLoading` can only be true for buttons with boundaries (ie. filled and outline buttons)
  - Note: There is currently no `button-outline-destructive` variant
- `invariant` throws dev side error on build if `isLoading` is set to true for ghost or unstyled variants
- Accessibility
  - `useReducedMotion` to [limit animations](https://www.framer.com/api/motion/guide-accessibility/)
  - `VisuallyHidden` for meaningful [loading](https://reach.tech/visually-hidden/) state change on screen readers

All comments and full commit history [here](https://github.com/AirLabsTeam/air-core/pull/69)

[Figma](https://www.figma.com/file/EZ1hzsmsNYblYoBQYvVgEV/Web-Component-Library?node-id=26%3A336) Design spec

### CODE REVIEW

File: packages/zephyr/src/Button.tsx [hunk 1]
```diff
@@ -26,19 +82,33 @@ export const Button = forwardRefWithAs<NonSemanticButtonProps, 'button'>(
...
+    const shouldReduceMotion = useReducedMotion();
 
     return (
       <Box
         as={as}
-        disabled={disabled}
+        disabled={disabled || isLoading}
         type={type}
         variant={variant}
+        className={isLoading ? `${className} isLoading` : `${className}`}
```
Comment (andyhqtran) [inline packages/zephyr/src/Button.tsx:111]: Can you use the classNames package here? So it looks like
```
import classNames from 'classnames';
...
className={classNames({ 'is-loading': isLoading }, className)}
```

**Comment by andyhqtran** [comment]
Can we also changing the cursor for the loading state?

File: packages/zephyr/src/Button.tsx [hunk 2]
```diff
@@ -154,6 +154,9 @@ export const Button = forwardRefWithAs<NonSemanticButtonProps, 'button'>(
           '&:disabled': {
             cursor: 'not-allowed',
           },
+          '&.isLoading': {
+            cursor: 'wait',
```
Comment (kylemh) [inline packages/zephyr/src/Button.tsx:158]: Hmmmm I straight up didn't know this was a thing. I guess we should ask design if they prefer `wait` or `progress`, so I can make that decision for SingleSelect too.

File: packages/zephyr/src/Button.tsx [hunk 3]
```diff
@@ -94,9 +94,11 @@ export const Button = forwardRefWithAs<NonSemanticButtonProps, 'button'>(
       (isLoading &&
         (variant === 'button-filled-blue' ||
           variant === 'button-filled-grey' ||
-          variant === 'button-filled-destructive')) ||
+          variant === 'button-filled-destructive' ||
+          variant === 'button-outline-blue' ||
+          variant === 'button-outline-grey')) ||
```
Comment (kylemh) [inline packages/zephyr/src/Button.tsx:99]: what about outline-destructive?

Comment (lin-jessica) [reply packages/zephyr/src/Button.tsx:99]: we don't have one in design

Comment (kylemh) [reply packages/zephyr/src/Button.tsx:99]: dont forget line 22

**Review by kylemh — APPROVED** [review]

**Pull request merged**